### PR TITLE
Fix Trashbin Expected Failures

### PR DESCRIPTION
**Pull request by refs**
| test | fixed? |
|-------------------------|-------------------------|
| apiTrashbin/trashbinFilesFolders.feature:215 | ✅ |
| apiTrashbin/trashbinFilesFolders.feature:216 | ✅ |
| apiTrashbin/trashbinFilesFolders.feature:230 | ✅ |
| apiTrashbin/trashbinFilesFolders.feature:231 | ✅ |

[Fixes on Reva](https://github.com/cs3org/reva/pull/1891)

| test | fixed? |
|-------------------------|-------------------------|
| apiTrashbin/trashbinSharingToShares.feature:23 | ✅ |
| apiTrashbin/trashbinSharingToShares.feature:24 | ✅ |
| apiShareManagementToShares/acceptShares.feature:490 | ✅ |
| apiShareManagementToShares/acceptShares.feature:491 | ✅ |

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Core-API-Tests-ocis-storage-8</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Core-API-Tests-ocis-storage-8</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Core-API-Tests-ocis-storage-8</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Core-API-Tests-ocis-storage-8</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Core-API-Tests-owncloud-storage-8</strong> failed. The build is cancelled...</summary>

File: proxy/pkg/webdav/response.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import (
+	"encoding/xml"
+	"net/http"
+)
+
+type code int
+
+const (
+	// SabredavBadRequest maps to HTTP 400
+	SabredavBadRequest code = iota
```
Comment (butonic) [inline proxy/pkg/webdav/response.go:12]: we are duplicating code from reva here, right? cant we reuse that?

**Review by butonic — CHANGES_REQUESTED** [review]

File: proxy/pkg/webdav/response.go (hunk 1 above)
Comment (refs) [reply proxy/pkg/webdav/response.go:12]: nope, it's under the internal package...

Comment (refs) [reply proxy/pkg/webdav/response.go:12]: [here](https://github.com/cs3org/reva/blob/f8b91e12b75329371e48b0d11c99c6416f0623b0/internal/http/services/owncloud/ocdav/error.go)

Comment (refs) [reply proxy/pkg/webdav/response.go:12]: We could however abstract the webdav response handling to an external package, but that's out of the scope of this PR IMHO. I can open a ticket as a "good first issue" sort of tasks

File: proxy/pkg/middleware/basic_auth.go [hunk 2]
```diff
@@ -61,6 +63,22 @@ func BasicAuth(optionSetters ...Option) func(next http.Handler) http.Handler {
...
+						b, err := webdav.Marshal(webdav.Exception{
+							Code:    webdav.SabredavPermissionDenied,
+							Message: "Authentication error",
+						})
+
+						webdav.HandleWebdavError(w, b, err)
+						return
+					}
+
 					w.WriteHeader(http.StatusUnauthorized)
```
Comment (C0rby) [inline proxy/pkg/middleware/basic_auth.go:82]: This line is redundant, isn't it?

Comment (refs) [reply proxy/pkg/middleware/basic_auth.go:82]: No, it still runs if the path is not a webdav request

Comment (refs) [reply proxy/pkg/middleware/basic_auth.go:82]: Totally

**Review by butonic — APPROVED** [review]

**Pull request merged**